drivers/wifi/nrf700x/zephyr/src/zephyr_wifi_mgmt.c

```diff
@@ -796,3 +796,73 @@ int wifi_nrf_set_power_save_timeout(const struct device *dev,
 out:
 	return ret;
 }
+
+#ifdef CONFIG_WIFI_FEAT_RAW_SCAN_RESULTS
+void wifi_nrf_rx_bcn_prb_resp_frm(void *vif_ctx,
+				  void *nwb,
+				  unsigned short frequency,
+				  signed short signal)
+{
+        struct wifi_nrf_vif_ctx_zep *vif_ctx_zep = vif_ctx;
+	struct wifi_nrf_ctx_zep *rpu_ctx_zep = NULL;
+	struct wifi_nrf_fmac_dev_ctx *fmac_dev_ctx = NULL;
+	struct wifi_raw_scan_result *bcn_prb_resp_info = NULL;
+	int frame_length = 0;
+	int val = signal;
+
+	vif_ctx_zep = vif_ctx;
+
+	if (!vif_ctx_zep) {
+		LOG_ERR("%s: vif_ctx_zep is NULL\n", __func__);
+		return;
+	}
+
+	if (!vif_ctx_zep->scan_in_progresss) {
+		return;
+	}
+
+	rpu_ctx_zep = vif_ctx_zep->rpu_ctx_zep;
+	if (!rpu_ctx_zep) {
+		LOG_ERR("%s: rpu_ctx_zep is NULL\n", __func__);
+		return;
+	}
+
+	fmac_dev_ctx = rpu_ctx_zep->rpu_ctx;
+
+	frame_length = wifi_nrf_osal_nbuf_data_size(fmac_dev_ctx->fpriv->opriv,
+						    nwb);
+
+	if (frame_length > CONFIG_WIFI_RAW_SCAN_RESULT_LENGTH) {
+		bcn_prb_resp_info = wifi_nrf_osal_mem_zalloc(
+					fmac_dev_ctx->fpriv->opriv,
+					sizeof(*bcn_prb_resp_info) +
+					CONFIG_WIFI_RAW_SCAN_RESULT_LENGTH);
+		memcpy(bcn_prb_resp_info->data,
+		       wifi_nrf_osal_nbuf_data_get(
+			       fmac_dev_ctx->fpriv->opriv,
+			       nwb),
+		       CONFIG_WIFI_RAW_SCAN_RESULT_LENGTH);
+
+	} else {
+		bcn_prb_resp_info = wifi_nrf_osal_mem_zalloc(
+					fmac_dev_ctx->fpriv->opriv,
+					sizeof(*bcn_prb_resp_info) +
+					frame_length);
+		memcpy(bcn_prb_resp_info->data,
+		       wifi_nrf_osal_nbuf_data_get(
+			       fmac_dev_ctx->fpriv->opriv,
+			       nwb),
+		       frame_length);
+	}
+
+	bcn_prb_resp_info->rssi = (val / 100);
```

Comment:
Please replace 100 by a macro and add a comment as to why this division by 100 is needed